### PR TITLE
Rename ServiceClusterReference to Region

### LINKED ISSUE #326: Rename ServiceClusterReference to Region

### PR DESCRIPTION
**Pull request by jiachengxu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #326 

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Rename ServiceClusterReference to Region
```

### CODE REVIEW

**Comment by nmiculinic** [comment]
It's looking good for me now, let @thetechnick have a look though

File: pkg/apis/catalog/v1alpha1/crds_types.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ type CRDInformation struct {
 	Plural   string       `json:"plural"`
 	Versions []CRDVersion `json:"versions"`
 
-	// ServiceCluster references a ServiceClusterReference of this CRD.
+	// ServiceCluster references a Region of this CRD.
 	ServiceCluster ObjectReference `json:"serviceCluster"`
```
Comment (thetechnick) [inline pkg/apis/catalog/v1alpha1/crds_types.go]:
```suggestion
	// Region references a Region of this CRD.
	Region ObjectReference `json:"region"`
```

**Pull request merged**